### PR TITLE
[Transform] Support "offset" parameter in DateHistogramGroupSource

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR adds support for the `offset` parameter in `DateHistogramGroupSource`.
`DateHistogramGroupSource.offset` gets translated directly to `date_histogram` aggregations's `offset` parameter which is described in the [Date Histogram Aggregation documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html#search-aggregations-bucket-datehistogram-offset).

Relates https://github.com/elastic/elasticsearch/issues/86452

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java [hunk 1]
```diff
@@ -197,26 +199,41 @@ private void writeInterval(Interval anInterval, StreamOutput out) throws IOExcep
...
 
     private static final ConstructingObjectParser<DateHistogramGroupSource, Void> STRICT_PARSER = createParser(false);
     private static final ConstructingObjectParser<DateHistogramGroupSource, Void> LENIENT_PARSER = createParser(true);
 
     private final Interval interval;
     private final ZoneId timeZone;
     private final Rounding.Prepared rounding;
-
-    public DateHistogramGroupSource(String field, ScriptConfig scriptConfig, boolean missingBucket, Interval interval, ZoneId timeZone) {
+    private final Long offset;
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java:210]: nit: wouldn't it be simpler to use a primitive `long` which defaults to `0`?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java [hunk 2]
```diff
@@ -231,6 +248,9 @@ private Rounding.Prepared buildRounding() {
         if (timeZone != null) {
             roundingBuilder.timeZone(timeZone);
         }
+        if (offset != 0) {
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java:251]: nit: the if looks superflous

**Review by hendrikmuhs — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSource.java:251]: Done.

**Review by przemekwitek — COMMENTED** [review]
Done.

**Pull request merged**